aspnet/fundamentals/error-handling.rst

```diff
@@ -1,8 +1,181 @@
+
+.. code-block:: c#
+
+	app.UseStatusCodePages(context => 
+		context.HttpContext.Response.SendAsync("Handler, status code: " +
+		context.HttpContext.Response.StatusCode, "text/plain"));
+
+Alternately, you can simply pass it a content type and a format string:
+
+.. code-block:: c#
+
+	app.UseStatusCodePages("text/plain", "Response, status code: {0}");
+
+The middleware can handle redirects (with either relative or absolute URL paths), passing the status code as part of the URL:
+
+.. code-block:: c#
+
+	app.UseStatusCodePagesWithRedirects("~/errors/{0}");
+
+In the above case, the client browser will see a ``302 / Found`` status and will redirect to the URL provided.
+
+Alternately, the middleware can re-execute the request from a new path format string:
+
+.. code-block:: c#
+
+	app.UseStatusCodePagesWithReExecute("/errors/{0}");
+
+The ``UseStatusCodePagesWithReExecute`` method will still return the original status code to the browser, but will also execute the handler given at the path specified.
+
+If you need to disable status code pages for certain requests, you can do so using the following code:
+
+.. code-block:: c#
+
+	var statusCodePagesFeature = context.Features.Get<IStatusCodePagesFeature>();
+	if (statusCodePagesFeature != null)
+	{
+		statusCodePagesFeature.Enabled = false;
+	}
+
+Limitations of Error Handling During Client-Server Interaction
+--------------------------------------------------------------
+
+ASP.NET apps have certain limitations to their error handling capabilities because of the nature of disconnected HTTP requests and responses. Keep these in mind as you design your app's error handling behavior.
+
+#. Once the headers for a response have been sent, you cannot change the response's status code.
+#. If the client disconnects mid-response, you cannot send them the rest of the content of that response.
+#. There is always the possibility of an error occuring one layer below your error handling layer.
+#. Don't forget, error handling pages can have errors, too. It's often a good idea for production error pages to consist of purely static content.
+
+Following the above recommendations will help ensure your app remains responsive and is able to gracefully handle errors that may occur.
+
+Server Error Handling
+---------------------
+
+In addition to ASP.NET, the server hosting your app will perform some error handling. In the case of IIS, this is described `here <https://technet.microsoft.com/en-us/library/cc731570(v=ws.10).aspx>`. For Kestrel, its built-in error handling behavior may be found `here <https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs#L565>`. Requests that are not handled by your app will be handled by the server, and any error that occurs will be handled by the server's error handling. Any custom error pages or error handling middleware or filters you have configured for your app will not affect this behavior.
```

Comment:
We should document the error handling behavior in Kestrel instead of pointing to the source code.